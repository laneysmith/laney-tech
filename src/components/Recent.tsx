import React from 'react';

import Section from './Shared/Section';
import { List, ListItem } from './Shared/List';

const Recent: React.FC = () => (
  <Section title="Recent activity">
    Lately I&apos;ve been...
    <List>
      {/* TODO: fix covid map */}
      {/* <ListItem $icon="ðŸ—º">
        building a data-driven map of <a href="https://covid.laney.tech/">Covid-19 cases</a> in US
        counties,
      </ListItem> */}
      <ListItem $icon="ðŸš´">
        mapping <a href="https://denver-bikes.laney.tech/">Denver&apos;s bike infrastructure</a>,
      </ListItem>
      <ListItem $icon="ðŸ§ª">
        conducting various <a href="https://github.com/laneysmith/">experiments</a>,
      </ListItem>
      <ListItem $icon="âœ¨">
        and building modern websites that make people&apos;s lives easier.
      </ListItem>
    </List>
  </Section>
);

export default Recent;
