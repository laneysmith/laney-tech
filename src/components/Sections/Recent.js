import React from 'react';

import SectionTemplate from './SectionTemplate';
import List from './List';
import ListItem from './ListItem';

const Recent = () => (
  <SectionTemplate title="Recent Activity">
    Lately I've been...
    <List>
      <ListItem icon="ðŸ—º">
        making <a href="https://covid.laney.tech/">data-driven maps</a>,
      </ListItem>
      <ListItem icon="ðŸ§ª">
        conducting various <a href="https://github.com/laneysmith/">experiments</a>,
      </ListItem>
      <ListItem icon="ðŸ““">
        writing{' '}
        <a href="https://dev.to/laney/mapbox-how-to-conditionally-style-features-based-on-covid-19-data-h78">
          more
        </a>{' '}
        <a href="https://dev.to/laney/react-mapbox-beginner-tutorial-2e35">articles</a>,
      </ListItem>
      <ListItem icon="âœ¨">and building modern websites that make people's lives easier.</ListItem>
    </List>
  </SectionTemplate>
);

export default Recent;
