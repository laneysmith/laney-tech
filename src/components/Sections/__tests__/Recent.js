import React from 'react';
import { render } from '@testing-library/react';

import Recent from '../Recent';

describe('Recent', () => {
  it('should match snapshot', () => {
    const { container } = render(<Recent />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
