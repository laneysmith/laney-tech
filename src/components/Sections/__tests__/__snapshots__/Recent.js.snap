// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent should match snapshot 1`] = `
.c0 {
  margin: 2em auto;
  padding: 0 1em;
  max-width: 550px;
}

.c1 {
  padding: 1em;
  margin-left: 1em;
  border-left: 1px solid;
}

.c2 {
  margin: 0.5em 0 0.5em 0.5em;
}

.c3 {
  display: inline-block;
  width: 20px;
  margin-right: 0.5em;
}

@media only screen and (max-width:600px) {
  .c0 {
    padding: 0 0.5em;
  }
}

<section
  class="c0"
>
  <h2>
    Recent Activity
  </h2>
  <div
    class="c1"
  >
    Lately I've been...
    <div
      class="c2"
      role="list"
    >
      <div
        role="listitem"
      >
        <span
          class="c3"
          role="img"
        >
          ðŸ—º
        </span>
        making 
        <a
          href="https://covid.laney.tech/"
        >
          data-driven maps
        </a>
        ,
      </div>
      <div
        role="listitem"
      >
        <span
          class="c3"
          role="img"
        >
          ðŸ§ª
        </span>
        conducting various 
        <a
          href="https://github.com/laneysmith/"
        >
          experiments
        </a>
        ,
      </div>
      <div
        role="listitem"
      >
        <span
          class="c3"
          role="img"
        >
          ðŸ““
        </span>
        writing a few 
        <a
          href="https://dev.to/laney"
        >
          articles
        </a>
         about coding,
      </div>
      <div
        role="listitem"
      >
        <span
          class="c3"
          role="img"
        >
          âœ¨
        </span>
        and building modern websites that make people's lives easier.
      </div>
    </div>
  </div>
</section>
`;
